internal/cucumber/scenario/scenario.go

```diff
@@ -0,0 +1,48 @@
+package scenario
+
+import (
+	"io/fs"
+
+	gherkin "github.com/cucumber/gherkin/go/v28"
+	messages "github.com/cucumber/messages/go/v24"
+	"github.com/rs/zerolog/log"
+)
+
+// List parses the provided files and returns a list of scenarios.
+func List(sys fs.FS, files []string) []*messages.Pickle {
+	uuid := &messages.UUID{}
+
+	var scenarios []*messages.Pickle
+	for _, filename := range files {
+		f, err := sys.Open(filename)
+		if err != nil {
+			log.Warn().Str("filename", filename).Msgf("Failed to open the file: %v", err)
+			continue
+		}
+		defer f.Close()
```

Comment:
🐛 Code smell! `defer` in a for-loop is discouraged, as the `defer` statement will only run after the entire loop has exited.

You might want to a create a separate method that does the gherkin doc parsing, and close the file in there.